test/Intl/NumberFormatOptions.js

```diff
@@ -82,18 +82,20 @@ var tests = [
         name: "Test Valid Options - Formatting Currency with Significant Digits",
         body: function () {
             const usdBaseOptions = { minimumSignificantDigits: 3, maximumSignificantDigits: 3, style: "currency", currency: "USD" };
-            var expectedRegex1 = new RegExp("\\$[\x20\u00a0]?123");
-            var expectedRegex2 = new RegExp("USD[\x20\u00a0]?123");
+            const symbolFormat = /\$[\x20\u00a0]?123/;
+            const codeFormat = /USD[\x20\u00a0]?123/;
+            const nameFormat = /(?:USD[\x20\u00a0]?123|123 US dollars)/; // ICU has a "long" format, WinGlob falls back to currency
+            const n = 123.1;
 
-            var actual1 = new Intl.NumberFormat("en-US", extendOptions(usdBaseOptions, {})).format(123.1);
-            assert.isTrue(expectedRegex1.test(actual1), "currency: USD (default display) actual: '" + actual1 + "' expected(Regex): '" + expectedRegex1 + "'");
-            var actual2 = new Intl.NumberFormat("en-US", extendOptions(usdBaseOptions, { currencyDisplay: "code" })).format(123.1);
-            assert.isTrue(expectedRegex2.test(actual2), "currency code: USD actual: '" + actual2 + "' expected(Regex): '" + expectedRegex2 + "'");
+            const defaultActual = new Intl.NumberFormat("en-US", extendOptions(usdBaseOptions, {})).format(n);
+            assert.isTrue(symbolFormat.test(defaultActual), `USD (default) actual: '${defaultActual}', which did not match ${symbolFormat.source}`);
+            const codeActual = new Intl.NumberFormat("en-US", extendOptions(usdBaseOptions, { currencyDisplay: "code" })).format(n);
+            assert.isTrue(codeFormat.test(codeActual), `USD (code) actual: '${codeActual}', which did not match ${codeFormat.source}`);
```

Comment:
Maybe it's too niche (IMO not really) but -- if we end up reusing this text-match-regex pattern (which I think we will) maybe UnitTestFramework could use an `assert.matches`. Just something to keep in mind for your next change like this.

Reply:
I like it, and will definitely consider it next time I have to edit a regex assert.

Reply:
When you do, note that both core and full tests will likely have baseline updates due to function IDs and/or line numbers in JS tests for stack traces and debug output.